ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -303,40 +339,19 @@ public SchemaKGroupedStream groupBy(
       );
     }
 
-    // Collect the column indexes, and build the new key as <column1>+<column2>+...
-    StringBuilder aggregateKeyName = new StringBuilder();
-    List<Integer> newKeyIndexes = new ArrayList<>();
-    boolean addSeparator = false;
-    for (Expression groupByExpr : groupByExpressions) {
-      if (addSeparator) {
-        aggregateKeyName.append("|+|");
-      } else {
-        addSeparator = true;
-      }
-      aggregateKeyName.append(groupByExpr.toString());
-      newKeyIndexes.add(
-          SchemaUtil.getIndexInSchema(groupByExpr.toString(), getSchema()));
-    }
+    final Pair<String, List<Integer>> aggregateKeyNameAndNewKeyIndexes =
```

Comment:
Personally, I'm not a fan of `Pair` type classes as it makes the code hard to read.  Concrete specific types lead to much more readable code IMHO.

Reply:
I'll just split this into 2 functions